### PR TITLE
- Added methods to NCubeController that were previously added to runt…

### PR DESCRIPTION
**Pull request by jsnyder4**
…ime and mutable interfaces

### CODE REVIEW

File: src/main/groovy/com/cedarsoftware/controller/NCubeController.groovy [hunk 1]
```diff
@@ -243,12 +260,19 @@ class NCubeController implements NCubeConstants, RpmVisualizerConstants
         return getJson(appId, cubeName, [mode:"json-index"])
     }
 
+    String getCubeRawJson(ApplicationID appId, String cubeName)
+    {
+        appId = addTenant(appId)
+        String rawJson = mutableClient.getCubeRawJson(appId, cubeName)
+        return rawJson
+    }
+
```
Comment (jdereg) [inline src/main/groovy/com/cedarsoftware/controller/NCubeController.groovy]: The NCubeManager call that this makes, does not call assertPermissions().  That makes this a security issue - anyone could call getCubeRawJson() and fetch cubes.  Please add an assertPermissions to NCubeManager.

File: src/main/groovy/com/cedarsoftware/controller/NCubeController.groovy [hunk 2]
```diff
@@ -1056,18 +1080,25 @@ class NCubeController implements NCubeConstants, RpmVisualizerConstants
         return cube.testData.toArray()
     }
 
+    Map getAppTests(ApplicationID appId)
+    {
+        appId = addTenant(appId)
+        Map appTests = mutableClient.getAppTests(appId)
+        return appTests
+    }
+
```
Comment (jdereg) [inline src/main/groovy/com/cedarsoftware/controller/NCubeController.groovy]: Same security issue - NCubeManager is not asserting permissions - meaning this call can be made restfully and no checks happen.

File: src/main/groovy/com/cedarsoftware/controller/NCubeController.groovy [hunk 3]
```diff
@@ -1501,6 +1532,12 @@ class NCubeController implements NCubeConstants, RpmVisualizerConstants
         return result
     }
 
+    NCube obsoletePullRequest(String prId)
+    {
+        NCube result = mutableClient.obsoletePullRequest(prId)
+        return result
+    }
+
```
Comment (jdereg) [inline src/main/groovy/com/cedarsoftware/controller/NCubeController.groovy]: no permissions check

File: src/main/groovy/com/cedarsoftware/controller/NCubeController.groovy (hunk 2 above)
Comment (tympollack) [inline src/main/groovy/com/cedarsoftware/controller/NCubeController.groovy]: getAppTests does run permissions checks on each cube it gets back from the persister.

File: src/main/groovy/com/cedarsoftware/controller/NCubeController.groovy (hunk 1 above)
Comment (tympollack) [inline src/main/groovy/com/cedarsoftware/controller/NCubeController.groovy]: Correct, the only place it was used before was after trying to loadCube (which does check permission), so it should be added in getCubeRawJson.

File: src/main/groovy/com/cedarsoftware/controller/NCubeController.groovy (hunk 3 above)
Comment (tympollack) [inline src/main/groovy/com/cedarsoftware/controller/NCubeController.groovy]: Note that permissions check has to happen here. On the manager, this is run as a system command which only happens when a PR merge encounters an illegal state, so it should not be called directly.

**Comment by jdereg** [comment]
thanks for getting all the remote tests working again

**Pull request merged**